---
title: "OPE_and_Covariate_Preprocessing_SexStratified"
author: "Samantha Lapehn"
date: "2023-10-23"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## OPE and Covariate Preprocessing for sex stratification

This code is related to the ECHO PATHWAYS project on orgranophosphate esters and the placental transcriptome in the CANDLE study. This code will perform exploratory analysis and preprocessing of the OPE exposure data and covariates. RNAseq preprocessing will be performed in a separate file. 

Covariate and Exposure data was received from the ECHO PATHWAYS data team on June 15, 2023 as part of data request P0106. 

OPE data was collected at M1, M2, or M3 visits for each individual, but are being assessed as a single dataset here.  

Here I will save the Covariate and Exposure data as separate Male and Female datasets for sex stratification.

### Load Packages
```{r load packages}
library(tidyverse)
library(corrplot)
library(stats)
library(ggvenn)
```

### Load Data
Will also load RNAseq data her since it contains the sequencing batch variable which needs formatted to be added to the covariates
```{r load data}
Exposure_and_Covariates <- read.csv("/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/Data_received061523/p0106_data.csv")
load(file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CANDLE_RNAseq/candle_20200226.Rdata")
```

### Separate Covariate Data
Gestational Age (h_birth_gestage) won't be included in our model, but including it so that it can be in the covariate table. 
```{r pull covariates}
Covariates_AllParticipants <- Exposure_and_Covariates %>%
  dplyr::select(pathways_id, h_c_sex, h_firstborn, h_birth_gestage, h_m_prepreg_bmi, h_m_race, h_m_enroll_educ, h_m_delivery_age, h_m_alcohol_prenatal, h_m_enroll_smoke, cg_labor_type, h_del_method, h_gravidity, h_placental_abrupt, cotinine_m1, cotinine_yn_m1, cotinine_lod_m1, cotinine_yn_m2)

#Split into separate dataframes based on child sex
Covariates_MaleParticipants_All <- Covariates_AllParticipants %>%
  dplyr::filter(h_c_sex=="M")

Covariates_FemaleParticipants_All <- Covariates_AllParticipants %>%
  dplyr::filter(h_c_sex=="F")
```

### Separate Exposure Data
```{r pull exposure data}
Exposures_AllParticipants <- Exposure_and_Covariates %>%
 dplyr::select(-cohort, -site, -h_c_sex, -h_firstborn, -h_birth_gestage, -h_m_prepreg_bmi, -h_m_race, -h_m_enroll_educ, -h_m_delivery_age, -h_m_alcohol_prenatal, -h_m_enroll_smoke, -cg_labor_type, -h_del_method, -h_gravidity, -h_placental_abrupt, -cotinine_m1, -cotinine_yn_m1, -cotinine_lod_m1, -cotinine_yn_m2, -child_birth_year)

#Split into separate dataframes based on child sex
Exposures_MaleParticipants <- Exposures_AllParticipants %>%
  dplyr::filter(pathways_id %in% Covariates_MaleParticipants_All$pathways_id)

Exposures_FemaleParticipants <- Exposures_AllParticipants %>%
  dplyr::filter(pathways_id %in% Covariates_FemaleParticipants_All$pathways_id)
```

### Batch Variable Preparation
```{r batch var prep}
sample_id <- colnames(candle_lengthScaledTPM$counts)
batches <- batches$Analysis
batch_var <- data.frame(sample_id, batches)
#check that they translated to the correct order in new DF
check1 <- batch_var$sample_id == colnames(candle_lengthScaledTPM$counts)
sum(check1)
#order is correct- will use this dataframe to combine with the other covariates that will be used in the analysis

```
### Filter to include only participants with RNAseq data
```{r RNAseq Filter}
Covariates_Male_RNAseqParticipants <- inner_join(Covariates_MaleParticipants_All, batch_var, by=c("pathways_id"="sample_id"))

Covariates_Female_RNAseqParticipants <- inner_join(Covariates_FemaleParticipants_All, batch_var, by=c("pathways_id"="sample_id"))

Exposures_Male_RNAseqParticipants <- Exposures_MaleParticipants %>%
  dplyr::filter(Exposures_MaleParticipants$pathways_id %in% Covariates_Male_RNAseqParticipants$pathways_id)

Exposures_Female_RNAseqParticipants <- Exposures_FemaleParticipants %>%
  dplyr::filter(Exposures_FemaleParticipants$pathways_id %in% Covariates_Female_RNAseqParticipants$pathways_id)
```

### Covariates- Remove NAs and partipants with exclusion criteria
Exclusion criteria = placental abruption
Also removing any participants with missing covariates
```{r Covariates Remove NAs}
Covariates_Male_RNAseqParticipants_NoNAs <- na.omit(Covariates_Male_RNAseqParticipants) 
Covariates_Female_RNAseqParticipants_NoNAs <- na.omit(Covariates_Female_RNAseqParticipants) 

Covariates_Male_Clean <- Covariates_Male_RNAseqParticipants_NoNAs %>%
  dplyr::filter(h_placental_abrupt==0)
Covariates_Female_Clean <- Covariates_Female_RNAseqParticipants_NoNAs %>%
  dplyr::filter(h_placental_abrupt==0)

#Next will update categorical variables to be factors for easier summarization
#First for Males
Covariates_Male_Clean$h_c_sex <- as.factor(Covariates_Male_Clean$h_c_sex)
Covariates_Male_Clean$h_firstborn <- as.factor(Covariates_Male_Clean$h_firstborn)
Covariates_Male_Clean$h_del_method <- as.factor(Covariates_Male_Clean$h_del_method)
Covariates_Male_Clean$h_m_race <- as.factor(Covariates_Male_Clean$h_m_race)
Covariates_Male_Clean$h_m_enroll_educ <- as.factor(Covariates_Male_Clean$h_m_enroll_educ)
Covariates_Male_Clean$h_m_alcohol_prenatal <- as.factor(Covariates_Male_Clean$h_m_alcohol_prenatal)
Covariates_Male_Clean$cg_labor_type <- as.factor(Covariates_Male_Clean$cg_labor_type)
Covariates_Male_Clean$batches <- as.factor(Covariates_Male_Clean$batches)

#Then for Females
Covariates_Female_Clean$h_c_sex <- as.factor(Covariates_Female_Clean$h_c_sex)
Covariates_Female_Clean$h_firstborn <- as.factor(Covariates_Female_Clean$h_firstborn)
Covariates_Female_Clean$h_del_method <- as.factor(Covariates_Female_Clean$h_del_method)
Covariates_Female_Clean$h_m_race <- as.factor(Covariates_Female_Clean$h_m_race)
Covariates_Female_Clean$h_m_enroll_educ <- as.factor(Covariates_Female_Clean$h_m_enroll_educ)
Covariates_Female_Clean$h_m_alcohol_prenatal <- as.factor(Covariates_Female_Clean$h_m_alcohol_prenatal)
Covariates_Female_Clean$cg_labor_type <- as.factor(Covariates_Female_Clean$cg_labor_type)
Covariates_Female_Clean$batches <- as.factor(Covariates_Female_Clean$batches)
```

### Add OPFR Gestage Variables to Covariates
OPFR_Gestage will be used as an interaction term, but will not be included as a confounding or precision variable.
```{r Combine OPFR Gestage Variable}
## Add OPFR Gestage Variables from exposure data
OPFR_Gestage_DF_Male <- Exposures_Male_RNAseqParticipants %>%
  dplyr::select(pathways_id, ofpr_gestage_m1, ofpr_gestage_m2)
OPFR_Gestage_DF_Male[is.na(OPFR_Gestage_DF_Male)] <- 0
OPFR_Gestage_DF_Male$OPFR_Gestage <- OPFR_Gestage_DF_Male$ofpr_gestage_m1 + OPFR_Gestage_DF_Male$ofpr_gestage_m2

OPFR_Gestage_DF_Female <- Exposures_Female_RNAseqParticipants %>%
  dplyr::select(pathways_id, ofpr_gestage_m1, ofpr_gestage_m2)
OPFR_Gestage_DF_Female[is.na(OPFR_Gestage_DF_Female)] <- 0
OPFR_Gestage_DF_Female$OPFR_Gestage <- OPFR_Gestage_DF_Female$ofpr_gestage_m1 + OPFR_Gestage_DF_Female$ofpr_gestage_m2

## Combine with other covariates
Covariates_Male_Clean <- left_join(Covariates_Male_Clean, OPFR_Gestage_DF_Male, by=c("pathways_id")) %>%
dplyr::select(-ofpr_gestage_m1, -ofpr_gestage_m2, -h_placental_abrupt)
Covariates_Female_Clean <- left_join(Covariates_Female_Clean, OPFR_Gestage_DF_Female, by=c("pathways_id")) %>%
dplyr::select(-ofpr_gestage_m1, -ofpr_gestage_m2, -h_placental_abrupt)
```

### Create a composite smoking variable
Using self-reported smoking and M1/M2 cotinine y/n variables to create a composite smoking variable where smoking will be designated if smoking is included at any of these three other variables. 
```{r cotinine prep}
Covariates_Male_Clean$Smoking_Composite <-rowSums(Covariates_Male_Clean[,c("h_m_enroll_smoke", "cotinine_yn_m1", "cotinine_yn_m2")])

Covariates_Female_Clean$Smoking_Composite <-rowSums(Covariates_Female_Clean[,c("h_m_enroll_smoke", "cotinine_yn_m1", "cotinine_yn_m2")])

#If Smoking Composite is 0 then non-smoking, if >0 then smoking

Covariates_Male_Clean$Smoking_Composite<-recode(Covariates_Male_Clean$Smoking_Composite, '0'="0", '1'="1", '2'="1", '3'="1")

Covariates_Female_Clean$Smoking_Composite<-recode(Covariates_Female_Clean$Smoking_Composite, '0'="0", '1'="1", '2'="1", '3'="1")

Covariates_Male_Clean$Smoking_Composite <- as.factor(Covariates_Male_Clean$Smoking_Composite)
summary(Covariates_Male_Clean$Smoking_Composite)

Covariates_Female_Clean$Smoking_Composite <- as.factor(Covariates_Female_Clean$Smoking_Composite)
summary(Covariates_Female_Clean$Smoking_Composite)

```

### Combine OPE Specific Gravity Variables
We need to combine the M1 and M2 specific gravity variables into one variable since we are treated these separate timepoints together. There is no overlap of participants between M1 and M2. Specific gravity will be used to adjust OPEs as part of the model.
```{r SG prep}

#Create list of M1 and M2 participants
M1Participants_Male <- OPFR_Gestage_DF_Male %>%
  dplyr::filter(ofpr_gestage_m1 >0) %>%
  dplyr::select(pathways_id)
M1Participants_Female <- OPFR_Gestage_DF_Female %>%
  dplyr::filter(ofpr_gestage_m1 >0) %>%
  dplyr::select(pathways_id)

M2Participants_Male <- OPFR_Gestage_DF_Male %>%
  dplyr::filter(ofpr_gestage_m2 >0) %>%
  dplyr::select(pathways_id)
M2Participants_Female <- OPFR_Gestage_DF_Female %>%
  dplyr::filter(ofpr_gestage_m2 >0) %>%
  dplyr::select(pathways_id)

#Create a dataframe with just the SG variables we need to use

SG_DF_Male <- Exposures_Male_RNAseqParticipants %>% 
  dplyr::select(pathways_id, ofpr_sg_m1, ofpr_sg_m2)
SG_DF_Female <- Exposures_Female_RNAseqParticipants %>% 
  dplyr::select(pathways_id, ofpr_sg_m1, ofpr_sg_m2)

#Use list of M1 and M2 participants to create separate SG dataframes then create new SG variable that will be combined based on participant study visit once re-combined

SG_DF_M1participants_Male <- SG_DF_Male %>%
  dplyr::filter(pathways_id %in% M1Participants_Male$pathways_id)
SG_DF_M1participants_Male$opfr_sg <- SG_DF_M1participants_Male$ofpr_sg_m1

SG_DF_M1participants_Female <- SG_DF_Female %>%
  dplyr::filter(pathways_id %in% M1Participants_Female$pathways_id)
SG_DF_M1participants_Female$opfr_sg <- SG_DF_M1participants_Female$ofpr_sg_m1

SG_DF_M2participants_Male <- SG_DF_Male %>%
  dplyr::filter(pathways_id %in% M2Participants_Male$pathways_id)
SG_DF_M2participants_Male$opfr_sg <- SG_DF_M2participants_Male$ofpr_sg_m2

SG_DF_M2participants_Female <- SG_DF_Female %>%
  dplyr::filter(pathways_id %in% M2Participants_Female$pathways_id)
SG_DF_M2participants_Female$opfr_sg <- SG_DF_M2participants_Female$ofpr_sg_m2
#Combine cotinine measures back into single data frame

SG_DF_New_Male <- full_join(SG_DF_M1participants_Male, SG_DF_M2participants_Male) %>%
  dplyr::select(-ofpr_sg_m1, -ofpr_sg_m2)

SG_DF_New_Female <- full_join(SG_DF_M1participants_Female, SG_DF_M2participants_Female) %>%
  dplyr::select(-ofpr_sg_m1, -ofpr_sg_m2)

# Add Specific Gravity to Clean Covariates
Covariates_Male_Clean <- inner_join(Covariates_Male_Clean, SG_DF_New_Male, by=c("pathways_id"))
Covariates_Female_Clean <- inner_join(Covariates_Female_Clean, SG_DF_New_Female, by=c("pathways_id"))
```


### Prepare Covariate Summary Statistics
Saving the covariate summary statistics to make the covariates tables.
```{r prep covariate summary statistics}
#Step 1: Summarize
summary(Covariates_Male_Clean)
summary(Covariates_Female_Clean)

#Step 2: Save as a dataframe to use for making table
Covar_Summary_DF_Male <- data.frame(unclass(summary(Covariates_Male_Clean)))
Covar_Summary_DF_Female <- data.frame(unclass(summary(Covariates_Female_Clean)))
#Step 3: Export dataframes to make summary tables
#write.csv(Covar_Summary_DF_Male, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/Covariate_Summaries/Covar_Summary_Male_102323.csv")
#write.csv(Covar_Summary_DF_Female, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/Covariate_Summaries/Covar_Summary_Female_102323.csv")
```

### Combine M1 and M2 exposure variables
Right now each OPE has two separate columns for M1 and M2 measurements. Need to combine these into a single column since there are no M1 and M2 overlaps
```{r exposure combining}
#First split data into a few dataframes
Exposure_Binary_LOD_vars_M1_Male <- Exposures_Male_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_lod_m1, dprp_lod_m1, dbup_dibp_lod_m1, bboep_lod_m1, behp_lod_m1, bcetp_lod_m1, bcpp_lod_m1, bdcpp_lod_m1, dphp_lod_m1, bmpp_lod_m1)

Exposure_Binary_LOD_vars_M1_Female <- Exposures_Female_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_lod_m1, dprp_lod_m1, dbup_dibp_lod_m1, bboep_lod_m1, behp_lod_m1, bcetp_lod_m1, bcpp_lod_m1, bdcpp_lod_m1, dphp_lod_m1, bmpp_lod_m1)

Exposure_Binary_LOD_vars_M2_Male <- Exposures_Male_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_lod_m2, dprp_lod_m2, dbup_dibp_lod_m2, bboep_lod_m2, behp_lod_m2, bcetp_lod_m2, bcpp_lod_m2, bdcpp_lod_m2, dphp_lod_m2, bmpp_lod_m2)
Exposure_Binary_LOD_vars_M2_Female <- Exposures_Female_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_lod_m2, dprp_lod_m2, dbup_dibp_lod_m2, bboep_lod_m2, behp_lod_m2, bcetp_lod_m2, bcpp_lod_m2, bdcpp_lod_m2, dphp_lod_m2, bmpp_lod_m2)

LOD_colnames <- c("pathways_id", "dep_lod", "dprp_lod", "dbup_dibp_lod", "bboep_lod", "behp_lod", "bcetp_lod", "bcpp_lod", "bdcpp_lod", "dphp_lod", "bmpp_lod")

colnames(Exposure_Binary_LOD_vars_M1_Male) <- LOD_colnames
colnames(Exposure_Binary_LOD_vars_M1_Female) <- LOD_colnames
colnames(Exposure_Binary_LOD_vars_M2_Male) <- LOD_colnames
colnames(Exposure_Binary_LOD_vars_M2_Female) <- LOD_colnames

Exposure_Binary_LOD_vars_M1_Male <- na.omit(Exposure_Binary_LOD_vars_M1_Male)
Exposure_Binary_LOD_vars_M1_Female <- na.omit(Exposure_Binary_LOD_vars_M1_Female)

Exposure_Binary_LOD_vars_M2_Male <- na.omit(Exposure_Binary_LOD_vars_M2_Male)
Exposure_Binary_LOD_vars_M2_Female <- na.omit(Exposure_Binary_LOD_vars_M2_Female)

Exposure_Binary_LOD_Combined_Male <- rbind(Exposure_Binary_LOD_vars_M1_Male, Exposure_Binary_LOD_vars_M2_Male)
Exposure_Binary_LOD_Combined_Female <- rbind(Exposure_Binary_LOD_vars_M1_Female, Exposure_Binary_LOD_vars_M2_Female)

Exposure_Binary_LOD_Combined_CompleteCovar_Male <- Exposure_Binary_LOD_Combined_Male %>%
  dplyr::filter(pathways_id %in% Covariates_Male_Clean$pathways_id)
Exposure_Binary_LOD_Combined_CompleteCovar_Female <- Exposure_Binary_LOD_Combined_Female %>%
  dplyr::filter(pathways_id %in% Covariates_Female_Clean$pathways_id)


Exposures_ImputedConcentrations_Male <- Exposures_Male_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_m1, dprp_m1, dbup_dibp_m1, bboep_m1, behp_m1, bcetp_m1, bcpp_m1, bdcpp_m1, dphp_m1, bmpp_m1, dep_m2, dprp_m2, dbup_dibp_m2, bboep_m2, behp_m2, bcetp_m2, bcpp_m2, bdcpp_m2, dphp_m2, bmpp_m2)
Exposures_ImputedConcentrations_Female <- Exposures_Female_RNAseqParticipants %>%
  dplyr::select(pathways_id, dep_m1, dprp_m1, dbup_dibp_m1, bboep_m1, behp_m1, bcetp_m1, bcpp_m1, bdcpp_m1, dphp_m1, bmpp_m1, dep_m2, dprp_m2, dbup_dibp_m2, bboep_m2, behp_m2, bcetp_m2, bcpp_m2, bdcpp_m2, dphp_m2, bmpp_m2)

#For the concentrations dataframe: 
#1- Turn NAs to zeros
#2- For each OPE then sum the m1 and m2 columns to create a single column for that OPE
#3- Create new dataframe that contains only the new m1 and m2 combined OPE concentration values
Exposures_ImputedConcentrations_Male[is.na(Exposures_ImputedConcentrations_Male)] <- 0
Exposures_ImputedConcentrations_Male$dep <- Exposures_ImputedConcentrations_Male$dep_m1 + Exposures_ImputedConcentrations_Male$dep_m2
Exposures_ImputedConcentrations_Male$dprp <- Exposures_ImputedConcentrations_Male$dprp_m1 + Exposures_ImputedConcentrations_Male$dprp_m2
Exposures_ImputedConcentrations_Male$dbup_dibp <- Exposures_ImputedConcentrations_Male$dbup_dibp_m1 + Exposures_ImputedConcentrations_Male$dbup_dibp_m2
Exposures_ImputedConcentrations_Male$bboep <- Exposures_ImputedConcentrations_Male$bboep_m1 + Exposures_ImputedConcentrations_Male$bboep_m2
Exposures_ImputedConcentrations_Male$behp <- Exposures_ImputedConcentrations_Male$behp_m1 + Exposures_ImputedConcentrations_Male$behp_m2
Exposures_ImputedConcentrations_Male$bcetp <- Exposures_ImputedConcentrations_Male$bcetp_m1 + Exposures_ImputedConcentrations_Male$bcetp_m2
Exposures_ImputedConcentrations_Male$bcpp <- Exposures_ImputedConcentrations_Male$bcpp_m1 + Exposures_ImputedConcentrations_Male$bcpp_m2
Exposures_ImputedConcentrations_Male$bdcpp <- Exposures_ImputedConcentrations_Male$bdcpp_m1 + Exposures_ImputedConcentrations_Male$bdcpp_m2
Exposures_ImputedConcentrations_Male$dphp <- Exposures_ImputedConcentrations_Male$dphp_m1 + Exposures_ImputedConcentrations_Male$dphp_m2
Exposures_ImputedConcentrations_Male$bmpp <- Exposures_ImputedConcentrations_Male$bmpp_m1 + Exposures_ImputedConcentrations_Male$bmpp_m2

Exposures_ImputedConcentrations_Female[is.na(Exposures_ImputedConcentrations_Female)] <- 0
Exposures_ImputedConcentrations_Female$dep <- Exposures_ImputedConcentrations_Female$dep_m1 + Exposures_ImputedConcentrations_Female$dep_m2
Exposures_ImputedConcentrations_Female$dprp <- Exposures_ImputedConcentrations_Female$dprp_m1 + Exposures_ImputedConcentrations_Female$dprp_m2
Exposures_ImputedConcentrations_Female$dbup_dibp <- Exposures_ImputedConcentrations_Female$dbup_dibp_m1 + Exposures_ImputedConcentrations_Female$dbup_dibp_m2
Exposures_ImputedConcentrations_Female$bboep <- Exposures_ImputedConcentrations_Female$bboep_m1 + Exposures_ImputedConcentrations_Female$bboep_m2
Exposures_ImputedConcentrations_Female$behp <- Exposures_ImputedConcentrations_Female$behp_m1 + Exposures_ImputedConcentrations_Female$behp_m2
Exposures_ImputedConcentrations_Female$bcetp <- Exposures_ImputedConcentrations_Female$bcetp_m1 + Exposures_ImputedConcentrations_Female$bcetp_m2
Exposures_ImputedConcentrations_Female$bcpp <- Exposures_ImputedConcentrations_Female$bcpp_m1 + Exposures_ImputedConcentrations_Female$bcpp_m2
Exposures_ImputedConcentrations_Female$bdcpp <- Exposures_ImputedConcentrations_Female$bdcpp_m1 + Exposures_ImputedConcentrations_Female$bdcpp_m2
Exposures_ImputedConcentrations_Female$dphp <- Exposures_ImputedConcentrations_Female$dphp_m1 + Exposures_ImputedConcentrations_Female$dphp_m2
Exposures_ImputedConcentrations_Female$bmpp <- Exposures_ImputedConcentrations_Female$bmpp_m1 + Exposures_ImputedConcentrations_Female$bmpp_m2

Exposures_ImputedConcentrations_Combined_Male <- Exposures_ImputedConcentrations_Male %>%
  dplyr::select(pathways_id, dep, dprp, dbup_dibp, bboep, behp, bcetp, bcpp, bdcpp, dphp, bmpp) %>%  dplyr::filter(pathways_id != "c0000044") 
#removed rows that were missing data 
Exposures_ImputedConcentrations_Combined_Female <- Exposures_ImputedConcentrations_Female %>%
  dplyr::select(pathways_id, dep, dprp, dbup_dibp, bboep, behp, bcetp, bcpp, bdcpp, dphp, bmpp) %>%  dplyr::filter(pathways_id != "c0005306") %>%
dplyr::filter(pathways_id != "c0000018") 

Exposures_ImputedConcentrations_Combined_CompleteCovar_Male <- Exposures_ImputedConcentrations_Combined_Male %>%
  dplyr::filter(pathways_id %in% Covariates_Male_Clean$pathways_id)

Exposures_ImputedConcentrations_Combined_CompleteCovar_Female <- Exposures_ImputedConcentrations_Combined_Female %>%
  dplyr::filter(pathways_id %in% Covariates_Female_Clean$pathways_id)

#Combine the LOD and concentrations values into one dataframe
Combined_ExposureData_Complete_Male <- inner_join(Exposures_ImputedConcentrations_Combined_CompleteCovar_Male, Exposure_Binary_LOD_Combined_CompleteCovar_Male, by=c("pathways_id"))

Combined_ExposureData_Complete_Female <- inner_join(Exposures_ImputedConcentrations_Combined_CompleteCovar_Female, Exposure_Binary_LOD_Combined_CompleteCovar_Female, by=c("pathways_id"))

```

### Summarize OPE Concentrations
Summarize and Save OPE concentrations to be used in a table
```{r OPE Concentrations}
#Change some categorical variables into Factors
#First for Male
Combined_ExposureData_Complete_Male$dep_lod <- as.factor(Combined_ExposureData_Complete_Male$dep_lod)
Combined_ExposureData_Complete_Male$dprp_lod <- as.factor(Combined_ExposureData_Complete_Male$dprp_lod)
Combined_ExposureData_Complete_Male$dbup_dibp_lod <-as.factor(Combined_ExposureData_Complete_Male$dbup_dibp_lod)
Combined_ExposureData_Complete_Male$bboep_lod <-as.factor(Combined_ExposureData_Complete_Male$bboep_lod)
Combined_ExposureData_Complete_Male$behp_lod <-as.factor(Combined_ExposureData_Complete_Male$behp_lod)
Combined_ExposureData_Complete_Male$bcetp_lod <-as.factor(Combined_ExposureData_Complete_Male$bcetp_lod)
Combined_ExposureData_Complete_Male$bcpp_lod <-as.factor(Combined_ExposureData_Complete_Male$bcpp_lod)
Combined_ExposureData_Complete_Male$bdcpp_lod <-as.factor(Combined_ExposureData_Complete_Male$bdcpp_lod)
Combined_ExposureData_Complete_Male$dphp_lod <-as.factor(Combined_ExposureData_Complete_Male$dphp_lod)
Combined_ExposureData_Complete_Male$bmpp_lod <-as.factor(Combined_ExposureData_Complete_Male$bmpp_lod)

#Then for Female
Combined_ExposureData_Complete_Female$dep_lod <- as.factor(Combined_ExposureData_Complete_Female$dep_lod)
Combined_ExposureData_Complete_Female$dprp_lod <- as.factor(Combined_ExposureData_Complete_Female$dprp_lod)
Combined_ExposureData_Complete_Female$dbup_dibp_lod <-as.factor(Combined_ExposureData_Complete_Female$dbup_dibp_lod)
Combined_ExposureData_Complete_Female$bboep_lod <-as.factor(Combined_ExposureData_Complete_Female$bboep_lod)
Combined_ExposureData_Complete_Female$behp_lod <-as.factor(Combined_ExposureData_Complete_Female$behp_lod)
Combined_ExposureData_Complete_Female$bcetp_lod <-as.factor(Combined_ExposureData_Complete_Female$bcetp_lod)
Combined_ExposureData_Complete_Female$bcpp_lod <-as.factor(Combined_ExposureData_Complete_Female$bcpp_lod)
Combined_ExposureData_Complete_Female$bdcpp_lod <-as.factor(Combined_ExposureData_Complete_Female$bdcpp_lod)
Combined_ExposureData_Complete_Female$dphp_lod <-as.factor(Combined_ExposureData_Complete_Female$dphp_lod)
Combined_ExposureData_Complete_Female$bmpp_lod <-as.factor(Combined_ExposureData_Complete_Female$bmpp_lod)

#print summary of values
summary(Combined_ExposureData_Complete_Male)
summary(Combined_ExposureData_Complete_Female)
#save summary of values as data frame
Exposure_Summary_DF_Male <- data.frame(unclass(summary(Combined_ExposureData_Complete_Male)))
Exposure_Summary_DF_Female <- data.frame(unclass(summary(Combined_ExposureData_Complete_Female)))


#export summary of values to create manuscript table
#write.csv(Exposure_Summary_DF_Male, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/OPE_Exposure_Summary_Male_102323.csv")
#write.csv(Exposure_Summary_DF_Female, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/OPE_Exposure_Summary_Female_102323.csv")
```
### Log Transform and Plot all exposures
Using the ng/ml concentrations with imputed values for those under the LOD
```{r plot all OPE}
rownames(Combined_ExposureData_Complete_Male) <-Combined_ExposureData_Complete_Male$pathways_id
rownames(Combined_ExposureData_Complete_Female) <-Combined_ExposureData_Complete_Female$pathways_id

Exposures_ConcOnly_Male <- Combined_ExposureData_Complete_Male %>%
  dplyr::select(dep, dprp, dbup_dibp, bboep, behp, bcetp, bcpp, bdcpp, dphp, bmpp)
Exposures_ConcOnly_Female <- Combined_ExposureData_Complete_Female %>%
  dplyr::select(dep, dprp, dbup_dibp, bboep, behp, bcetp, bcpp, bdcpp, dphp, bmpp)

LogConcOnly_Male <- log(Exposures_ConcOnly_Male) #Using default settings here which are natural log (this is the same log transformation that was used for the PAH paper)
LogConcOnly_Female <- log(Exposures_ConcOnly_Female)

boxplot(LogConcOnly_Male, main="Male-Log OPE Concentrations (ng/mL)", cex.axis=0.7)
boxplot(LogConcOnly_Female, main="Female-Log OPE Concentrations (ng/mL)", cex.axis=0.7)
```


### LOD Cutoff
Requiring 70% to be greater than LOD. 
Need 0 = 253 or higher for Male
Need 0 = 264 of higher for Female
```{r LOD Cutoff}

summary(Combined_ExposureData_Complete_Male$dep_lod) #yes
summary(Combined_ExposureData_Complete_Male$dprp_lod) #no
summary(Combined_ExposureData_Complete_Male$dbup_dibp_lod) #yes
summary(Combined_ExposureData_Complete_Male$bboep_lod) #no
summary(Combined_ExposureData_Complete_Male$behp_lod) #no
summary(Combined_ExposureData_Complete_Male$bcetp_lod) #yes
summary(Combined_ExposureData_Complete_Male$bcpp_lod) #no
summary(Combined_ExposureData_Complete_Male$bdcpp_lod) #yes
summary(Combined_ExposureData_Complete_Male$dphp_lod) #yes
summary(Combined_ExposureData_Complete_Male$bmpp_lod) #no

summary(Combined_ExposureData_Complete_Female$dep_lod) #yes
summary(Combined_ExposureData_Complete_Female$dprp_lod) #no
summary(Combined_ExposureData_Complete_Female$dbup_dibp_lod) #yes
summary(Combined_ExposureData_Complete_Female$bboep_lod) #no
summary(Combined_ExposureData_Complete_Female$behp_lod) #no
summary(Combined_ExposureData_Complete_Female$bcetp_lod) #yes
summary(Combined_ExposureData_Complete_Female$bcpp_lod) #no
summary(Combined_ExposureData_Complete_Female$bdcpp_lod) #yes
summary(Combined_ExposureData_Complete_Female$dphp_lod) #yes
summary(Combined_ExposureData_Complete_Female$bmpp_lod) #no
```



# Plot only exposure data with 70% >LOD
```{r plot greater than LOD cutoff}
Exposures_ConcOnly_LODCutoff_Male <- Combined_ExposureData_Complete_Male %>%
  dplyr::select(dep, dbup_dibp, bcetp, bdcpp, dphp)
Exposures_ConcOnly_LODCutoff_Female <- Combined_ExposureData_Complete_Female %>%
  dplyr::select(dep, dbup_dibp, bcetp, bdcpp, dphp)

Exposures_LogConcOnly_LODCutoff_Male <- log(Exposures_ConcOnly_LODCutoff_Male) #Log Transformation using default settings for natural log transformation (this is the same log transformation that was used for PAH data)
Exposures_LogConcOnly_LODCutoff_Female <- log(Exposures_ConcOnly_LODCutoff_Female)

boxplot(Exposures_LogConcOnly_LODCutoff_Male, main="Male-Log OPE Concentrations (ng/mL) (70% > LOD)", cex.axis=0.7)
boxplot(Exposures_LogConcOnly_LODCutoff_Female, main="Female-Log OPE Concentrations (ng/mL) (70% > LOD)", cex.axis=0.7)

```

# Correlation Plot (Pearson Correlation)
Plotting the correlation of OPEs that meet the 70%>LOD cutoff using a pearson correlation (linear correlation)
Using raw (not log adjusted) OPE values for these correlations
```{r correlation plots}
corr_pearson_Male <- cor(Exposures_ConcOnly_LODCutoff_Male)
corrplot(corr_pearson_Male, method="color")
print(corr_pearson_Male)

corr_pearson_Female <- cor(Exposures_ConcOnly_LODCutoff_Female)
corrplot(corr_pearson_Female, method="color")
print(corr_pearson_Female)
```

### Correlation Plot (Spearman Correlation)
Plotting the correlation of OPEs that meet the 70%>LOD cutoff using a spearman correlation (nonparametric correlation)
Using raw (not log adjusted) OPE values for these correlations
```{r correlation plots Spearman}

corr_spear_Male <- cor(Exposures_ConcOnly_LODCutoff_Male, method="spearman")
corrplot(corr_spear_Male, method="color")
print(corr_spear_Male)

corr_spear_Female <- cor(Exposures_ConcOnly_LODCutoff_Female, method="spearman")
corrplot(corr_spear_Female, method="color")
print(corr_spear_Female)

```

### PCA- Male
Running a principal components analysis to assess variation in the data
Only including OPEs that meet the 70% over the LOD threshold
```{r PCA Male}
PCA_matrix_Male <-as.matrix(Exposures_LogConcOnly_LODCutoff_Male)
PC_object_Male<-prcomp(PCA_matrix_Male, scale=TRUE)
PCs_Male <- PC_object_Male$x

PCs_Male =as.data.frame(PCs_Male[,1:5])

EVAL_Male<-as.data.frame(matrix(NA,nrow=5,ncol=5))
colnames(EVAL_Male)<-c("P_DEP","P_DBUP_DIBP", "P_BCETP", "P_BDCPP", "P_DPHP")
PC_nums_Male <-c("1", "2", "3", "4", "5")

for (i in 1:5){
  AOV<-aov(PCs_Male[,i]~dep,data=Exposures_LogConcOnly_LODCutoff_Male)
  EVAL_Male[i,1] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Male[,i]~dbup_dibp,data=Exposures_LogConcOnly_LODCutoff_Male)
  EVAL_Male[i,2] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Male[,i]~bcetp,data=Exposures_LogConcOnly_LODCutoff_Male)
  EVAL_Male[i,3] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Male[,i]~bdcpp,data=Exposures_LogConcOnly_LODCutoff_Male)
  EVAL_Male[i,4] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Male[,i]~dphp,data=Exposures_LogConcOnly_LODCutoff_Male)
  EVAL_Male[i,5] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

#Proportion of Variation Explained by Each PC
EVAL_Male <-cbind(EVAL_Male, PC_nums_Male)
EVAL_Male$PropVar <-summary(PC_object_Male)$importance["Proportion of Variance", 1:5]

ggplot(PCs_Male, aes(x=PC1, y=PC2)) +
  geom_point()+
  ggtitle("PC1 and PC2")

EVAL_Male %>%
  ggplot(aes(x=PC_nums_Male, y=PropVar, group=1)) +
  geom_line()+
  geom_point() +
  ggtitle("Male-Variance by PC") + 
  ylab("Proportion of Variance") +
  ylim(0, 1)

print(EVAL_Male)
```

### PCA- Female
Running a principal components analysis to assess variation in the data
Only including OPEs that meet the 70% over the LOD threshold
```{r PCA Female}
PCA_matrix_Female <-as.matrix(Exposures_LogConcOnly_LODCutoff_Female)
PC_object_Female<-prcomp(PCA_matrix_Female, scale=TRUE)
PCs_Female <- PC_object_Female$x

PCs_Female =as.data.frame(PCs_Female[,1:5])

EVAL_Female<-as.data.frame(matrix(NA,nrow=5,ncol=5))
colnames(EVAL_Female)<-c("P_DEP","P_DBUP_DIBP", "P_BCETP", "P_BDCPP", "P_DPHP")
PC_nums_Female <-c("1", "2", "3", "4", "5")

for (i in 1:5){
  AOV<-aov(PCs_Female[,i]~dep,data=Exposures_LogConcOnly_LODCutoff_Female)
  EVAL_Female[i,1] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Female[,i]~dbup_dibp,data=Exposures_LogConcOnly_LODCutoff_Female)
  EVAL_Female[i,2] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Female[,i]~bcetp,data=Exposures_LogConcOnly_LODCutoff_Female)
  EVAL_Female[i,3] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Female[,i]~bdcpp,data=Exposures_LogConcOnly_LODCutoff_Female)
  EVAL_Female[i,4] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

for (i in 1:5){
  AOV<-aov(PCs_Female[,i]~dphp,data=Exposures_LogConcOnly_LODCutoff_Female)
  EVAL_Female[i,5] <-summary(AOV)[[1]][["Pr(>F)"]][1]
}

#Proportion of Variation Explained by Each PC
EVAL_Female <-cbind(EVAL_Female, PC_nums_Female)
EVAL_Female$PropVar <-summary(PC_object_Female)$importance["Proportion of Variance", 1:5]

ggplot(PCs_Female, aes(x=PC1, y=PC2)) +
  geom_point()+
  ggtitle("Female-PC1 and PC2")

EVAL_Female %>%
  ggplot(aes(x=PC_nums_Female, y=PropVar, group=1)) +
  geom_line()+
  geom_point() +
  ggtitle("Female-Variance by PC") + 
  ylab("Proportion of Variance") +
  ylim(0, 1)

print(EVAL_Female)
```

### Plot Distribution of Study Visit Timing
This variable will be used as an interaction term to assess effect modification
```{r study visit timing}
hist(Covariates_Male_Clean$OPFR_Gestage, main="Male- Gestational Age at Study Visit")

hist(Covariates_Female_Clean$OPFR_Gestage, main="Female- Gestational Age at Study Visit")
```

# Make an Exposure Dataframe with only binary or continuous data for each OPE based on if it met the LOD threshold
```{r LOD threshold exposure categorization}
#Male
Concentrations_Male <- Combined_ExposureData_Complete_Male %>%
  dplyr::select(pathways_id, dep, dbup_dibp, bcetp, bdcpp, dphp)
rownames(Concentrations_Male) <- Concentrations_Male$pathways_id
Concentrations_Male <- Concentrations_Male[,-1]
LogConcentrationCont_Male <- log(Concentrations_Male)
LogConcentrationCont_Male$pathways_id <- rownames(LogConcentrationCont_Male)
colnames(LogConcentrationCont_Male) <- c("log_dep", "log_dbup_dibp", "log_bcetp", "log_bdcpp", "log_dphp", "pathways_id")

Combined_ExposureData_Complete_Male <- full_join(Combined_ExposureData_Complete_Male, LogConcentrationCont_Male, by=c("pathways_id"))

ExposureData_Binary_or_LogContinuous_Male <- Combined_ExposureData_Complete_Male %>%
  dplyr::select(pathways_id, log_dep, log_dbup_dibp, log_bcetp, log_bdcpp, log_dphp, dprp_lod, bboep_lod, behp_lod, bcpp_lod, bmpp_lod)

#Female

Concentrations_Female <- Combined_ExposureData_Complete_Female %>%
  dplyr::select(pathways_id, dep, dbup_dibp, bcetp, bdcpp, dphp)
rownames(Concentrations_Female) <- Concentrations_Female$pathways_id
Concentrations_Female <- Concentrations_Female[,-1]
LogConcentrationCont_Female <- log(Concentrations_Female)
LogConcentrationCont_Female$pathways_id <- rownames(LogConcentrationCont_Female)
colnames(LogConcentrationCont_Female) <- c("log_dep", "log_dbup_dibp", "log_bcetp", "log_bdcpp", "log_dphp", "pathways_id")

Combined_ExposureData_Complete_Female <- full_join(Combined_ExposureData_Complete_Female, LogConcentrationCont_Female, by=c("pathways_id"))

ExposureData_Binary_or_LogContinuous_Female <- Combined_ExposureData_Complete_Female %>%
  dplyr::select(pathways_id, log_dep, log_dbup_dibp, log_bcetp, log_bdcpp, log_dphp, dprp_lod, bboep_lod, behp_lod, bcpp_lod, bmpp_lod)
```

# Export Cleaned Covariates and Exposure Data
```{r export clean data}

#MALE

#write.csv(Covariates_Male_Clean, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanCovariates_Male_102323.csv")

#write.csv(Combined_ExposureData_Complete_Male, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanOPEData_Male_102323.csv")

#write.csv(ExposureData_Binary_or_LogContinuous_Male, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanOPEData_BinaryLogContinuous_Male_102323.csv")

#FEMALE

#write.csv(Covariates_Female_Clean, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanCovariates_Female_102323.csv")

#write.csv(Combined_ExposureData_Complete_Female, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanOPEData_Female_102323.csv")

#write.csv(ExposureData_Binary_or_LogContinuous_Female, "/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/CleanOPEData_BinaryLogContinuous_Female_102323.csv")
```








