---
title: "QGComp"
author: "Samantha Lapehn"
date: "2023-11-29"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## QGComp for OPE Data
Using this code to run quantile g-based computation for OPE mixtures and the placental transcriptome using code shared by Dennis Khodsevich https://github.com/D-Khodasevich/EWAS_QGComp

### Load Packages and Functions
```{r packages}
library(tidyverse)
library(qgcomp)
library(data.table)
library(edgeR)

source("/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/EWAS_QGComp Code from Dennis/EWAS_QGComp_Functions_Update3.R")
```

### Load Data
Need covariates and exposure data, as well as gene expression with genes as rows and samples as columns. 
```{r load data}
load(file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/FilteredNormalizedData_102023.RData")
```

### Prepare Exposure Data
Need to remove binary exposures so that only the continuous OPEs are included
```{r prep exposure data}
Mixture <- OPE_Data_Select_Ready %>%
  dplyr::select(log_dep, log_dbup_dibp, log_bcetp, log_bdcpp, log_dphp)
Mixture$pathways_id <-rownames(Mixture)

Xnm <- c("log_dep", "log_dbup_dibp", "log_bcetp", "log_bdcpp", "log_dphp")
```

### Prepare Covariate Data
Need to remove any covariates that we aren't using
```{r covar prep}
Covariates_Selected <- Covariates %>%
  dplyr::select(pathways_id, h_c_sex, opfr_sg, h_m_delivery_age, h_m_prepreg_bmi, h_m_race, h_m_enroll_educ, batches, cg_labor_type, h_del_method, h_m_alcohol_prenatal, Smoking_Composite, h_firstborn, h_gravidity)

covars <- c("h_c_sex", "opfr_sg", "h_m_delivery_age", "h_m_prepreg_bmi", "h_m_race", "h_m_enroll_educ", "batches", "cg_labor_type", "h_del_method", "h_m_alcohol_prenatal", "Smoking_Composite", "h_firstborn", "h_gravidity")
```

### Combine Covariate and Exposure Data into Single dataframe
```{r combine exposure and covar}
Pheno <- inner_join(Covariates_Selected, Mixture, by=c("pathways_id"))
rownames(Pheno) <- Pheno$pathways_id
Pheno <- Pheno %>%
  dplyr::select(-pathways_id)
class(Pheno$h_c_sex)
#Make sure certain covariates are indicated as factors
Pheno$h_c_sex <- as.factor(Pheno$h_c_sex)
Pheno$h_m_race <- as.factor(Pheno$h_m_race)
Pheno$h_m_enroll_educ <- as.factor(Pheno$h_m_enroll_educ)
Pheno$batches <- as.factor(Pheno$batches)
Pheno$cg_labor_type <- as.factor(Pheno$cg_labor_type)
Pheno$h_del_method<- as.factor(Pheno$h_del_method)
Pheno$h_m_alcohol_prenatal <- as.factor(Pheno$h_m_alcohol_prenatal)
Pheno$Smoking_Composite <- as.factor(Pheno$Smoking_Composite)
Pheno$h_firstborn <- as.factor(Pheno$h_firstborn)
Pheno$h_gravidity <- as.factor(Pheno$h_gravidity)

#Check data types
summary(Pheno)

```

### Prep Expression Data
```{r expr prep}
meth <- cpm(Y_norm$counts, log=T)
```

### Check that order of Expression and Pheno Data are the same
```{r check order}
check <- colnames(meth) == rownames(Pheno)
sum(check)
```

### Split for parallelization
To Parallelize the function need to split the meth matrix into smaller pieces. 
```{r split}
#Full Data
meth_DF <- data.frame(meth)
meth_list <- split(meth_DF, rep(1:1289, each=10))
meth_list_matrix <- lapply(meth_list, as.matrix)

#Test Data
meth_test <- head(meth_DF, 1000)
meth_test_list <-split(meth_test, rep(1:100, each=10))
meth_test_list_matrix <- lapply(meth_test_list, as.matrix)
```


### Save input data that will be needed to run Bootstrap QGComp on the terminal
```{r save input data}
#save(meth_list_matrix, file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/QGCompInputforTerminal/ExpressionData_List_120823.Rdata")

#save(Pheno, file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/QGCompInputforTerminal/CovariateData_120823.Rdata")

#save(covars, file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/QGCompInputforTerminal/CovariateNameVector_120823.Rdata")

#save(Xnm, file="/active/paquette_a/slapehn/OPE_and_PlacentalTranscriptome/Data/QGCompInputforTerminal/MixtureNameVector_120823.Rdata")
```
